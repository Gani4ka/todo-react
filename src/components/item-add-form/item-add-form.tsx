import React, { useState } from 'react';

import './item-add-form.css';

export interface ItemAddFormProps {
  onItemAdded(arg0: string): void;
}

export const ItemAddForm = (props: ItemAddFormProps): JSX.Element => {
  const { onItemAdded } = props;
  const [label, setLabel] = useState('');

  const onLabelChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setLabel(e.target.value);
  };

  const onSubmit = (e: React.FormEvent): void => {
    e.preventDefault();
    onItemAdded(label);
    setLabel('');
  };
  return (
    <form className='item-add-form d-flex' onSubmit={onSubmit}>
      <input
        type='text'
        className='form-control'
        onChange={onLabelChange}
        placeholder='What needs to be done'
        value={label}
      />
      <button className='btn btn-outline-secondary'>Add</button>
    </form>
  );
};
