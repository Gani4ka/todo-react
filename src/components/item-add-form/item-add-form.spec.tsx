import React from 'react';
import { shallow } from 'enzyme';
import { ItemAddForm, ItemAddFormProps } from './item-add-form';

describe('<ItemAddForm />', (): void => {
  const prop: ItemAddFormProps = {
    onItemAdded: () => {}
  };
  it('should render correctly', (): void => {
    const component = shallow(<ItemAddForm {...prop} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('should call handler', (): void => {
    const component = shallow(<ItemAddForm {...prop} />);
    component.find('.btn').simulate('click');
    expect(prop.onItemAdded).toBeCalled();
  });
});
