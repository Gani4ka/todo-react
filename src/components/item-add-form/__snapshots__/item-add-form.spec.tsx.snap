// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemAddForm /> should render correctly 1`] = `
"<form className=\\"item-add-form d-flex\\" onSubmit={[Function: onSubmit]}>
  <input type=\\"text\\" className=\\"form-control\\" onChange={[Function: onLabelChange]} placeholder=\\"What needs to be done\\" value=\\"\\" />
  <button className=\\"btn btn-outline-secondary\\">
    Add
  </button>
</form>"
`;
